scripts/nbd_client_manager.py

```diff
@@ -208,11 +208,16 @@ def disconnect_nbd_device(nbd_device):
     This function is idempotent: calling it on an already disconnected device
     does nothing.
     """
-    if _is_nbd_device_connected(nbd_device=nbd_device):
-        _remove_persistent_connect_info(nbd_device)
-        cmd = ['nbd-client', '-disconnect', nbd_device]
-        _call(cmd)
-        _wait_for_nbd_device(nbd_device=nbd_device, connected=False)
+    try:
+        if _is_nbd_device_connected(nbd_device=nbd_device):
```

Comment:
So this check doesn't always give the right answer?

Reply:
If the device isn't present at all it fails with an exception `NbdDeviceNotFound` so there are three states

- Connected
- Disconnected
- Not present